services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/TenantUserController.java

```diff
@@ -0,0 +1,263 @@
+import com.sourcefuse.jarc.services.usertenantservice.dto.UserView;
+import com.sourcefuse.jarc.services.usertenantservice.enums.AuthorizeErrorKeys;
+import com.sourcefuse.jarc.services.usertenantservice.enums.PermissionKey;
+import com.sourcefuse.jarc.services.usertenantservice.service.TenantUserService;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import jakarta.validation.Valid;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@Slf4j
+@RequestMapping("/tenants/{id}/users")
+@RequiredArgsConstructor
+public class TenantUserController {
+
+  private final TenantUserService tnUsrService;
+
+  @PersistenceContext
+  private EntityManager em;
+
+  @PostMapping("")
+  public ResponseEntity<Object> createUserTenants(
+    @Valid @RequestBody UserDto userDto,
+    @PathVariable("id") UUID id
+  ) {
+    if (id == null) {
+      throw new ResponseStatusException(
+        HttpStatus.BAD_REQUEST,
+        "${tenant.id.not.specified}"
+      );
+    }
+
+    userDto.setTenantId(id);
+    userDto
+      .getUserDetails()
+      .setEmail(userDto.getUserDetails().getEmail().toLowerCase());
+    userDto
+      .getUserDetails()
+      .setUsername(userDto.getUserDetails().getUsername().toLowerCase());
+    /** map created for ruling with options */
+    Map<String, String> option = new HashMap<>();
+    option.put("authId", userDto.getAuthId());
+    option.put("authProvider", userDto.getAuthProvider());
+    IAuthUserWithPermissions currentUser =
+      (IAuthUserWithPermissions) SecurityContextHolder
+        .getContext()
+        .getAuthentication()
+        .getPrincipal();
+    //tnUsrService.create(userDto, currentUser, option);
+
+    return new ResponseEntity<>("", HttpStatus.CREATED);
+  }
+
+  @GetMapping("")
+  public ResponseEntity<Object> gtUsrTntByRole(@PathVariable("id") UUID id) {
```

Comment:
function name is not readable
```suggestion
  public ResponseEntity<Object> gtUsrTntByRole(@PathVariable("id") UUID id) {
  public ResponseEntity<Object> getUserTenantByRole(@PathVariable("id") UUID id) {
```

Reply:
already done